platform/darwin/src/MGLStyleValue_Private.h

```diff
@@ -181,11 +181,16 @@ class MGLStyleValueTransformer {
         return mbglValue;
     }
     
-    NSObject* toRawStyleSpecValue(MGLColor *color) {
-        return @(color.mgl_color.stringify().c_str());
-    }
-    
-    NSObject* toRawStyleSpecValue(NSObject *rawMGLValue) {
+    /**
+     As hack to allow converting enum => string values, we accept a second, dummy parameter in
+     the toRawStyleSpecValue() methods for converting 'atomic' (non-style-function) values.
+     This allows us to use `std::enable_if` to test (at compile time) whether or not MBGLType is an Enum.
+     */
+    template <typename MBGLEnum = MBGLType,
+    class = typename std::enable_if<!std::is_enum<MBGLEnum>::value>::type,
+    typename MGLEnum = ObjCEnum,
+    class = typename std::enable_if<!std::is_enum<MGLEnum>::value>::type>
+    NSObject* toRawStyleSpecValue(NSObject *rawMGLValue, MBGLEnum &mbglValue) {
         if ([rawMGLValue isKindOfClass:[NSValue class]]) {
             const auto rawNSValue = (NSValue *)rawMGLValue;
             if (strcmp([rawNSValue objCType], @encode(CGVector)) == 0) {
```

Comment:
By the way, I see special handling for CGVector values, but not NSEdgeInsets/UIEdgeInsets values. Do we need it too? Here’s the [full correspondence](https://www.mapbox.com/ios-sdk/api/3.5.0-beta.1/for-style-authors.html#setting-attribute-values) between JSON and Foundation types.